src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -3132,9 +3132,9 @@ MarkAttributeDirty ColorControlServer::SetQuietReportAttribute(QuieterReportingA
 
     if (isStartOrEndOfTransition)
     {
-        // At the start or end of the movement/transition we must report
-        auto predicate = [](const typename QuieterReportingAttribute<Q>::SufficientChangePredicateCandidate &) -> bool {
-            return true;
+        // At the start or end of the movement/transition we must report if the value changed
+        auto predicate = [](const typename QuieterReportingAttribute<Q>::SufficientChangePredicateCandidate & candidate) -> bool {
+            return (candidate.lastDirtyValue != candidate.newValue);
```

Comment:
Predicates are only called `if (newValue != mLastDirtyValue)`, so I don't think this change is needed...   As in, in this predicate `candidate.lastDirtyValue != candidate.newValue` is in fact always `true`.

Reply:
Yes, you are right, it went over my head! I guess it is not needed in the provided predicate `GetPredicateForSufficientTimeSinceLastDirty` either